src/toxcore/dht/request_queue.rs

```diff
@@ -79,80 +93,69 @@ mod tests {
 
     #[test]
     fn clone() {
-        let queue = RequestQueue::new(Duration::from_secs(42));
+        let queue = RequestQueue::<()>::new(Duration::from_secs(42));
         let _ = queue.clone();
     }
 
     #[test]
     fn insert_new_ping_id() {
-        crypto_init().unwrap();
         let mut queue = RequestQueue::new(Duration::from_secs(42));
-        let (pk, _sk) = gen_keypair();
 
-        let ping_id = queue.new_ping_id(pk);
+        let ping_id = queue.new_ping_id(7);
 
-        assert!(queue.ping_map.contains_key(&(pk, ping_id)));
+        assert_eq!(queue.ping_map[&ping_id].1, 7);
     }
 
     #[test]
     fn check_ping_id() {
-        crypto_init().unwrap();
```

Comment:
gen_ping_id calls to random_u32 which requires crypto_init()?

Reply:
You are right, fixed.